juju/juju
# [JUJU-1427]rewrite apiserver client cloudtests to use gomock generated mocks

This PR replaces most of the old handwritten  mocks in the cloudfacade api unit tests with `GoMock` generated mocks.

## Checklist


## QA steps

```sh
go test -check.v -check.f cloudSuiteGomocks github.com/juju/juju/apiserver/facades/client/cloud/
```
